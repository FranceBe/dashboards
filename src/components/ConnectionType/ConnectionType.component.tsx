// ConnectionType component to display an icon matching
// The device connection type with a specific color
import Tooltip from '@material-ui/core/Tooltip'
import SettingsEthernetIcon from '@material-ui/icons/SettingsEthernet'
import SignalCellularAltIcon from '@material-ui/icons/SignalCellularAlt'
import WifiIcon from '@material-ui/icons/Wifi'
import { ConnectionTypeContainer } from 'components/ConnectionType/connectionType.style'
import React, { ReactElement } from 'react'
import { ConnectionTypeProps } from 'types/connectionType'

export const ConnectionType: React.FC<ConnectionTypeProps> = ({ connection_type }) => {
  const iconByType: Record<'ethernet' | 'wifi' | 'cellular', ReactElement> = {
    cellular: <SignalCellularAltIcon data-testid={'icon-cellular'} />,
    ethernet: <SettingsEthernetIcon data-testid={'icon-ethernet'} />,
    wifi: <WifiIcon data-testid={'icon-wifi'} />,
  }

  return (
    <ConnectionTypeContainer data-testid={'connection_type-container'} type={connection_type}>
      <Tooltip title={connection_type}>{iconByType[connection_type]}</Tooltip>
    </ConnectionTypeContainer>
  )
}

export default ConnectionType
