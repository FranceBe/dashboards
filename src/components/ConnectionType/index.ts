import ConnectionTypeComponent, {
  ConnectionType,
} from 'components/ConnectionType/ConnectionType.component'

export { ConnectionType }

export default ConnectionTypeComponent
