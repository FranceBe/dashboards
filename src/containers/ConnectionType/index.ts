import ConnectionTypeContainer, {
  ConnectionType,
} from 'containers/ConnectionType/ConnectionType.container'

export { ConnectionType }

export default ConnectionTypeContainer
