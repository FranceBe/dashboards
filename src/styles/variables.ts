// Colors to use matching the graphic chart
export const palette: { [key: string]: string } = {
  blue_light: '#f0f0fc',
  green_primary: '#65db25',
  grey_primary: '#343536',
  red_primary: '#eb4034',
}

// Spaces to use to respect an 8px grid
export const spaces: { [key: string]: string } = {
  large: '24px',
  medium: '16px',
  regular: '8px',
  small: '4px',
}

// Font sizes to use in project
export const fontSizes: { [key: string]: string } = {
  small: '12px',
}
