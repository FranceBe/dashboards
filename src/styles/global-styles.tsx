import { createGlobalStyle } from 'styled-components'
import { palette } from 'styles/variables'

export const GlobalStyle = createGlobalStyle`
  body {
    margin: 1.5em;
    background-color: ${palette.blue_light};
    color: ${palette.grey_primary};
  }
  a {
    text-decoration: none;
    :active, :visited {
      color: ${palette.blue_primary};
    }
    :hover, :visited:hover {
      color: ${palette.blue_secondary};
    }
  }
`
