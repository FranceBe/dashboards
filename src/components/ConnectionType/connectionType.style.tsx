import React from 'react'
import styled from 'styled-components'
import { fontSizes, palette, spaces } from 'styles/variables'

export const ConnectionTypeContainer = styled((props) => <div {...props} />)<{
  type: 'ethernet' | 'wifi' | 'cellular'
}>`
  background-color: ${(props) => {
    if (props.type === 'wifi') {
      return palette.purple_primary
    } else if (props.type === 'ethernet') {
      return palette.pink_primary
    } else {
      return palette.yellow_light
    }
  }};
  width: fit-content;
  border-radius: ${spaces.small};
  padding: ${spaces.regular};
  font-size: ${fontSizes.small};
`
