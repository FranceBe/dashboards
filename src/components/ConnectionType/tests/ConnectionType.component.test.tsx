import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { render, screen } from '@testing-library/react'
import ConnectionType from 'components/ConnectionType'
import React from 'react'
import { palette } from 'styles/variables'

describe('ConnectionType', () => {
  describe('Given connection_type is "wifi"', () => {
    it('should match snapshot', () => {
      const { container } = render(<ConnectionType connection_type={'wifi'} />)
      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have green_primary background color', () => {
      const { container } = render(<ConnectionType connection_type={'wifi'} />)
      expect(container.firstChild).toHaveStyleRule('background-color', palette.purple_secondary)
    })
    it('should find the connected icon', () => {
      render(<ConnectionType connection_type={'wifi'} />)
      expect(screen.getByTestId('icon-wifi')).toBeInTheDocument()
    })
  })
  describe('Given connection_type is "ethernet"', () => {
    it('should match snapshot', () => {
      const { container } = render(<ConnectionType connection_type={'ethernet'} />)
      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have green_primary background color', () => {
      const { container } = render(<ConnectionType connection_type={'ethernet'} />)
      expect(container.firstChild).toHaveStyleRule('background-color', palette.blue_primary)
    })
    it('should find the connected icon', () => {
      render(<ConnectionType connection_type={'ethernet'} />)
      expect(screen.getByTestId('icon-ethernet')).toBeInTheDocument()
    })
  })
  describe('Given connection_type is "cellular"', () => {
    it('should match snapshot', () => {
      const { container } = render(<ConnectionType connection_type={'cellular'} />)
      expect(container.firstChild).toMatchSnapshot()
    })
    it('should have green_primary background color', () => {
      const { container } = render(<ConnectionType connection_type={'cellular'} />)
      expect(container.firstChild).toHaveStyleRule('background-color', palette.green_secondary)
    })
    it('should find the connected icon', () => {
      render(<ConnectionType connection_type={'cellular'} />)
      expect(screen.getByTestId('icon-cellular')).toBeInTheDocument()
    })
  })
})
