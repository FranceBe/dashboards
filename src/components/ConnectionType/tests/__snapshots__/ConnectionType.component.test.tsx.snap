// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionType Given connection_type is "cellular" should match snapshot 1`] = `
.c0 {
  background-color: #003725;
  box-shadow: 0 5px 25px 0 rgba(0,0,45,0.1),0 2px 6px 0 rgba(0,0,0,0.04), 0 16px 24px 0 rgba(0,0,46,0.06);
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 4px;
  padding: 8px;
  font-size: 0.75rem;
}

.c0 svg {
  fill: #ffffff;
}

<div
  class="c0"
  data-testid="connection_type-container"
  type="cellular"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    data-testid="icon-cellular"
    focusable="false"
    title="cellular"
    viewBox="0 0 24 24"
  >
    <path
      d="M17 4h3v16h-3zM5 14h3v6H5zm6-5h3v11h-3z"
    />
  </svg>
</div>
`;

exports[`ConnectionType Given connection_type is "ethernet" should match snapshot 1`] = `
.c0 {
  background-color: #0E074B;
  box-shadow: 0 5px 25px 0 rgba(0,0,45,0.1),0 2px 6px 0 rgba(0,0,0,0.04), 0 16px 24px 0 rgba(0,0,46,0.06);
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 4px;
  padding: 8px;
  font-size: 0.75rem;
}

.c0 svg {
  fill: #ffffff;
}

<div
  class="c0"
  data-testid="connection_type-container"
  type="ethernet"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    data-testid="icon-ethernet"
    focusable="false"
    title="ethernet"
    viewBox="0 0 24 24"
  >
    <path
      d="M7.77 6.76L6.23 5.48.82 12l5.41 6.52 1.54-1.28L3.42 12l4.35-5.24zM7 13h2v-2H7v2zm10-2h-2v2h2v-2zm-6 2h2v-2h-2v2zm6.77-7.52l-1.54 1.28L20.58 12l-4.35 5.24 1.54 1.28L23.18 12l-5.41-6.52z"
    />
  </svg>
</div>
`;

exports[`ConnectionType Given connection_type is "wifi" should match snapshot 1`] = `
.c0 {
  background-color: #45327D;
  box-shadow: 0 5px 25px 0 rgba(0,0,45,0.1),0 2px 6px 0 rgba(0,0,0,0.04), 0 16px 24px 0 rgba(0,0,46,0.06);
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 4px;
  padding: 8px;
  font-size: 0.75rem;
}

.c0 svg {
  fill: #ffffff;
}

<div
  class="c0"
  data-testid="connection_type-container"
  type="wifi"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    data-testid="icon-wifi"
    focusable="false"
    title="wifi"
    viewBox="0 0 24 24"
  >
    <path
      d="M1 9l2 2c4.97-4.97 13.03-4.97 18 0l2-2C16.93 2.93 7.08 2.93 1 9zm8 8l3 3 3-3c-1.65-1.66-4.34-1.66-6 0zm-4-4l2 2c2.76-2.76 7.24-2.76 10 0l2-2C15.14 9.14 8.87 9.14 5 13z"
    />
  </svg>
</div>
`;
